File: IFMG - SJE/Testes/bhaskara.cpp
// PROGRAMA QUE CALCULA A FÓRMULA DE BHASKARA

#include <iostream>
#include <cmath>
#include <windows.h>

using namespace std;

struct Formula
{
    double a, b, c, xVertice, yVertice, cortaEixoY;
};

void limpaTela();
void leDoTeclado();
void calculaBhaskara();
void maisUmCalculo();
void msgFinal();

Formula bhaskara;
string opc = "S";
int cont = 0;

int main()
{
    SetConsoleOutputCP(CP_UTF8);
    system("cls");
    cout << "\n>>>>>>Fórmula de Bhaskara<<<<<<\n";
    while (true)
    {
        cont++;
        leDoTeclado();
        calculaBhaskara();
        maisUmCalculo();
        if (opc == "N" || opc == "n")
            break;
        limpaTela();
    }
    msgFinal();

    system("pause");
    return 0;
}

void msgFinal()
{
    if (cont == 1)
        cout << "\nRealizamos " << cont << " cálculo dessa vez!" << endl;
    else
        cout << "\nRealizamos " << cont << " cálculos dessa vez!" << endl;
    cout << "Volte sempre que precisar!" << endl;
}

void maisUmCalculo()
{
    while (true)
    {
        cout << "\nMais um cálculo: [S/N] ";
        cin >> opc;
        if (opc == "S" || opc == "s" || opc == "N" || opc == "n")
            break;
        cout << "Opção inválida! ";
    }
}

void limpaTela()
{
    system("cls");
    cout << "Vamos mais uma vez!" << "\n\n";
}

void calculaBhaskara()
{
    float delta, x1, x2;
    delta = pow(bhaskara.b, 2) - (4 * bhaskara.a * bhaskara.c);
    if (delta < 0)
    {
        cout << "\nDelta é igual a: " << delta << endl;
        cout << "Não existe raízes reais, pois o delta é menor que zero." << "\n\n";
    }
    else if (delta == 0)
    {
        x1 = (-bhaskara.b + sqrt(delta)) / (2 * bhaskara.a);
        cout << "\nEquação possui apenas uma raíz, pois o delta é igual a zero." << endl;
        cout << "A raíz é igual a: " << x1 << "\n\n";
    }
    else
    {
        x1 = (-bhaskara.b + sqrt(delta)) / (2 * bhaskara.a);
        x2 = (-bhaskara.b - sqrt(delta)) / (2 * bhaskara.a);
        cout << "\nDelta é igual a: " << delta << endl;
        cout << "X1 é igual a: " << x1 << endl;
        cout << "X2 é igual a: " << x2 << "\n\n";
    }
    bhaskara.cortaEixoY = pow(bhaskara.a * 0, 2) + 0 * bhaskara.b + bhaskara.c;
    cout("Intercepta o eixo Y no %.2f\n", bhaskara.cortaEixoY);
    bhaskara.xVertice = -bhaskara.b / 2 * bhaskara.a;
    bhaskara.yVertice = -delta / 4 * bhaskara.a;
    cout("X do vértice é: %.2f\n", bhaskara.xVertice);
    cout("Y do vértice é: %.2f\n", bhaskara.yVertice);
    if (bhaskara.a > 0)
    {
        cout("Como a = %.2f > 0, ponto de mínimo.\n", bhaskara.a);
        cout("Im = {Y e R | y >= %.2f}\n", bhaskara.yVertice);
    }
    else
    {
        cout("Como a = %.2f < 0, ponto de máximo.\n", bhaskara.a);
        cout("Im = {Y e R | y <= %.2f}\n", bhaskara.yVertice);
    }
}

void leDoTeclado()
{
    cout << "Insira o valor de A: ";
    cin >> bhaskara.a;
    cout << "Insira o valor de B: ";
    cin >> bhaskara.b;
    cout << "Insira o valor de C: ";
    cin >> bhaskara.c;
}

File: PERÍODOS FECHADOS/Treinos/bhaskara.py
# PROGRAMA QUE CALCULA A FÓRMULA DE BHASKARA

from math import pow ,sqrt
from time import sleep
print('=-' * 35)
print('Olá, eu sou um programa que calcula a Fórmula de Bhaskara para você!!')
print('=-' * 35)

cont = 1

while True:
    a = float(input('Insira o valor de A: '))
    b = float(input('Insira o valor de B: '))
    c = float(input('Insira o valor de C: '))
    print('Calculando...\n')
    sleep(2)
    delta = pow(b, 2) - (4 * a * c)
    if delta < 0:
        print(f'O Delta é igual a {delta}.')
        sleep(1)
        print('E por ele ser menor que zero, não existe raiz real!')
    elif delta == 0:
        x1 = ( -b + sqrt(delta)) / (2 * a)
        print(f"Equação possui apenas uma raiz, pois o delta é igual a zero.")
        print(f"A raiz é igual a: {x1}.")
    else:
        x1 = ( -b + sqrt(delta)) / (2 * a)
        x2 = (-b - sqrt(delta)) / (2 * a)
        print(f'O Delta é igual a {delta}.')
        sleep(1)
        print(f'X1 é igual a {x1}.')
        sleep(1)
        print(f'X2 é igual a {x2}.')
        sleep(1)
    opc = str(input('\nVocê quer continuar? [S/N] ')).strip().upper()[0]
    while opc not in 'SN':
        opc = str(input("Opção inválida! Digite novamente: [S/N] ")).strip().upper()[0]
    if opc in 'N':
        break
    cont+=1
print('=-' * 20)
if cont == 1:
    print(f'Realizamos {cont} cálculo hoje!')
else:
    print(f'Realizamos {cont} cálculos hoje!')
sleep(0.5)
print('Programa Encerrado!')
sleep(0.5)
print('Volte sempre que precisar!!')
